src/Hopper/Internal/Type/BinderInfo.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+module Hopper.Internal.Type.BinderInfo
+  ( BinderInfo(..), Type
+  ) where
+
+import Hopper.Internal.Type.Relevance (Relevance)
+
+import Data.Data (Data)
+import Data.Text (Text)
+import Data.Typeable (Typeable)
+import GHC.Generics (Generic)
+
+type Type = () -- FIXME: use real types
+
+data BinderInfo =
+  BinderInfoData { relevanceBI :: Relevance
+                 -- ^ if zero, during evaluation we need not pass around
+                 --  BUT during NORMALIZATION, we probably do
+                 --  so the normalizer WILL thread around irrelevant values
+                 -- to properly support dependent type checking
+                 -- as is proper, because a runtime irrelevant value
+                 -- SHOULD be relevant during type checking, or
+                 -- it has no reason to exist
+                 , typeBICC :: Type
```

Comment:
and/or we have that each "lowering" is a subset of the types in more restricted forms and maybe a few more extra assumed/builtin datatypes that are per se definable in userland